### PR TITLE
 regard conversion as method instead of constructor

### PR DESCRIPTION
**Pull request by FirstLoveLife**
1. According to http://eel.is/c++draft/class.ctor and http://eel.is/c++draft/#class, user-defined implicit conversion should be regarded as the constructor. 
2. 

```
case Decl::CXXConstructor:	   
    case Decl::CXXConversion:	
      func->def.kind = lsSymbolKind::Constructor;
```

will be weird in lsp-imenu:
```
(Class)
  ┃ operator bool (Constructor)
  ┃ operator& (Method)
  ┃ __bit_const_reference<_Cp> (Constructor)
  ┃ __bit_const_reference<_Cp> (Constructor)
  ┃ __storage_pointer (Other)
  ┃ __storage_type (Other)
  ...
```

Clearly, operator bool is not this class's `Constructor`. This commit changes the layout into:

```
(Class)
  ┃ operator bool (Method)
  ┃ operator& (Method)
  ┃ __bit_const_reference<_Cp> (Constructor)
  ┃ __bit_const_reference<_Cp> (Constructor)
  ┃ __storage_pointer (Other)
  ┃ __storage_type (Other)
  ...
```

### CODE REVIEW

File: src/indexer.cc [hunk 1]
```diff
@@ -1033,10 +1033,10 @@ class IndexDataConsumer : public index::IndexDataConsumer {
       break;
     }
     case Decl::CXXConstructor:
-    case Decl::CXXConversion:
       func->def.kind = lsSymbolKind::Constructor;
       break;
     case Decl::CXXDestructor:
```
Comment (MaskRay) [inline src/indexer.cc]: Can you swap them?

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ build
 debug
 release
 /compile_commands.json
+Release
```
Comment (MaskRay) [inline .gitignore]: It was a fault that I (or someone else) put debug/release in `.gitignore`

File: src/indexer.cc (hunk 1 above)
Comment (FirstLoveLife) [reply src/indexer.cc]: Ok

Comment (FirstLoveLife) [reply src/indexer.cc]: @MaskRay Updated

**Comment by MaskRay** [comment]
cherry picked "regard conversion as method instead of constructor". Thanks!

**Pull request closed without merging**